official/mnist/mnist.py

```diff
@@ -22,8 +22,11 @@
 import sys
 
 import tensorflow as tf
+from tensorflow.python.client import device_lib
```

Comment:
Could these public https://www.tensorflow.org/versions/master/api_docs/python/tf/contrib/eager/num_gpus   or   https://www.tensorflow.org/versions/master/api_docs/python/tf/Session#list_devices  serve as an alternative?

Reply:
Ah, interesting, I did not realize that. It seems like reaching into Eager would break some implicit rules here as well, and reaching into the session requires that we have the session available at the time. I am perfectly fine with a function-internal import, especially since we anticipate removing this functionality when it is native to the replication itself.